### PR TITLE
demo: update Grid、Table、Select demo

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

before：

image
image

after:

image
image

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | demo: update demo |
| 🇨🇳 Chinese | demo: update demo |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Review by crazyair — APPROVED** [review]

File: components/table/demo/jsx.tsx [hunk 1]
```diff
@@ -52,13 +52,19 @@ const App: React.FC = () => (
...
       render={(tags: string[]) => (
-        <>
-          {tags.map((tag) => (
-            <Tag color="blue" key={tag}>
-              {tag}
-            </Tag>
-          ))}
-        </>
+        <Flex gap="small">
+          {tags.map<React.ReactNode>((tag) => {
```
Comment (crazyair) [inline components/table/demo/jsx.tsx:56]: 上面已经设置了 `string[]` 这里不需要 React.ReactNode 吧？

还有其他的地方

Comment (li-jia-nan) [reply components/table/demo/jsx.tsx:56]: 不一样的，不是同一个地方

Comment (crazyair) [reply components/table/demo/jsx.tsx:56]: tags 是 string[]，所以 tag 是 string

Comment (li-jia-nan) [reply components/table/demo/jsx.tsx:56]: 这个 ReactNode 指的是 map 的返回值类型，不是 item 的类型

File: components/table/demo/basic.tsx [hunk 2]
```diff
@@ -32,8 +32,8 @@ const columns: TableProps<DataType>['columns'] = [
     key: 'tags',
     dataIndex: 'tags',
     render: (_, { tags }) => (
-      <>
-        {tags.map((tag) => {
+      <Flex gap="small">
+        {tags.map<React.ReactNode>((tag) => {
```
Comment (MadCcc) [inline components/table/demo/basic.tsx:36]: 这种指定类型感觉没啥必要，都是能自动 infer 出来的，demo 尽量保持简单的好

**Pull request merged**